File: utils/io/convert.py
import os
import pandas as pd
import scanpy as sc
import argparse
from rpy2 import robjects

parser = argparse.ArgumentParser()
parser.add_argument('filename')
args = parser.parse_args()

data = sc.read(args.filename)
data.X = data.raw.to_adata()[:, data.var_names].X
counts = data.to_df()
metadata = data.obs

metadata_tmp, counts_tmp = args.filename.split(".h5ad")[0]+"_metadata.csv", args.filename.split(".h5ad")[0]+"_counts.csv"
print(f"Saving tmp files to {metadata_tmp} and {counts_tmp}.")

counts.to_csv(counts_tmp)
metadata.to_csv(metadata_tmp)

File: utils/io/convert.R
require(Seurat)

args <- commandArgs(trailingOnly = TRUE)
filename <- args[1]
name <- sub('.h5ad', '', filename, fixed = TRUE)
counts_tmp = paste0(name,"_counts.csv")
metadata_tmp = paste0(name,"_metadata.csv")


seu <- CreateSeuratObject(counts=t(read.csv(counts_tmp, row.names=1)), assay="RNA")
seu@meta.data <- read.csv(metadata_tmp, row.names=1)

saveRDS(seu, paste0(name,".rds"))

file.remove(counts_tmp)
file.remove(metadata_tmp)